<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EFIBootEditor</class>
 <widget class="QMainWindow" name="EFIBootEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>EFI Boot Editor</string>
  </property>
  <property name="windowIcon">
   <iconset theme="preferences-system"></iconset>
  </property>
  <widget class="QWidget" name="boot">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0,2">
    <item>
     <widget class="BootEntryListView" name="entries_list">
      <property name="toolTip">
       <string>Boot entries</string>
      </property>
      <property name="statusTip">
       <string>Boot entries</string>
      </property>
      <property name="whatsThis">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This is the list of the boot entries.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustToContents</enum>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="horizontalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <property name="movement">
       <enum>QListView::Snap</enum>
      </property>
      <property name="resizeMode">
       <enum>QListView::Adjust</enum>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="entries_actions">
      <item>
       <widget class="QPushButton" name="entry_add">
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Add new entry</string>
        </property>
        <property name="statusTip">
         <string>Add new entry</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Click this to add new boot entry.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset theme="add"></iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="entry_remove">
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Remove entry</string>
        </property>
        <property name="statusTip">
         <string>Remove entry</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Click this to remove currently selected entry.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset theme="remove"></iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="entry_move_up">
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Move entry up</string>
        </property>
        <property name="statusTip">
         <string>Move entry up</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Click this to move currently selected entry up.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset theme="up"></iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="entry_move_down">
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Move entry down</string>
        </property>
        <property name="statusTip">
         <string>Move entry down</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Click this to move currently selected entry down.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset theme="down"></iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="entries_actions_spacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="settings">
      <item>
       <widget class="BootEntryForm" name="entry_form" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="toolTip">
         <string>Entry form</string>
        </property>
        <property name="statusTip">
         <string>Entry form</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Entry form&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="settings_spacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="global">
        <property name="toolTip">
         <string>Global settings</string>
        </property>
        <property name="statusTip">
         <string>Global settings</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Global settings&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="title">
         <string>Global settings</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <layout class="QFormLayout" name="global_form">
         <item row="0" column="0">
          <widget class="QLabel" name="timeout_label">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Boot manager timeout&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="statusTip">
            <string>Timeout</string>
           </property>
           <property name="text">
            <string>Timeout</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="timeout_number">
           <property name="toolTip">
            <string>Ttimeout</string>
           </property>
           <property name="statusTip">
            <string>Timeout</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Timeout&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Menu bar</string>
   </property>
   <property name="statusTip">
    <string>Menu bar</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Menu bar&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <widget class="QMenu" name="file">
    <property name="toolTip">
     <string>File</string>
    </property>
    <property name="statusTip">
     <string>File</string>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;File menu&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="save"/>
    <addaction name="export_"/>
    <addaction name="import_"/>
    <addaction name="reload"/>
    <addaction name="separator"/>
    <addaction name="exit"/>
   </widget>
   <widget class="QMenu" name="help">
    <property name="toolTip">
     <string>Help</string>
    </property>
    <property name="statusTip">
     <string>Help</string>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Help menu&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="about"/>
    <addaction name="separator"/>
    <addaction name="dump_raw_efi_data"/>
   </widget>
   <addaction name="file"/>
   <addaction name="help"/>
  </widget>
  <widget class="QStatusBar" name="status">
   <property name="toolTip">
    <string>Status bar</string>
   </property>
   <property name="statusTip">
    <string>Status bar</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Status bar&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <action name="exit">
   <property name="icon">
    <iconset theme="exit"></iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="statusTip">
    <string>Exit</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Exit the program&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="save">
   <property name="icon">
    <iconset theme="document-save"></iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="statusTip">
    <string>Save</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Apply changes to the system&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="reload">
   <property name="icon">
    <iconset theme="reload"></iconset>
   </property>
   <property name="text">
    <string>Reload</string>
   </property>
   <property name="statusTip">
    <string>Reload</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Reloads EFI data from the system&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="about">
   <property name="icon">
    <iconset theme="help"></iconset>
   </property>
   <property name="text">
    <string>About EFI Boot Editor</string>
   </property>
   <property name="statusTip">
    <string>About EFI Boot Editor</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show information about the program&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="export_">
   <property name="icon">
    <iconset theme="document-export"></iconset>
   </property>
   <property name="text">
    <string>Export</string>
   </property>
   <property name="statusTip">
    <string>Export</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Export current entries into JSON&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="import_">
   <property name="icon">
    <iconset theme="document-import"></iconset>
   </property>
   <property name="text">
    <string>Import</string>
   </property>
   <property name="statusTip">
    <string>Import</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Import EFI data from JSON dump&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="dump_raw_efi_data">
   <property name="icon">
    <iconset theme="document-export"></iconset>
   </property>
   <property name="text">
    <string>Dump raw EFI data</string>
   </property>
   <property name="statusTip">
    <string>Dump raw EFI data</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Dumps raw EFI data for debugging purposes&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BootEntryListView</class>
   <extends>QListView</extends>
   <header>bootentrylistview.h</header>
   <slots>
    <signal>selected(QModelIndex)</signal>
    <slot>removeCurrentRow()</slot>
    <slot>insertRow()</slot>
    <slot>moveCurrentRowUp()</slot>
    <slot>moveCurrentRowDown()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>BootEntryForm</class>
   <extends>QWidget</extends>
   <header>bootentryform.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>entry_move_up</sender>
   <signal>clicked()</signal>
   <receiver>entries_list</receiver>
   <slot>moveCurrentRowUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>entry_move_down</sender>
   <signal>clicked()</signal>
   <receiver>entries_list</receiver>
   <slot>moveCurrentRowDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>entry_add</sender>
   <signal>clicked()</signal>
   <receiver>entries_list</receiver>
   <slot>insertRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>entry_remove</sender>
   <signal>clicked()</signal>
   <receiver>entries_list</receiver>
   <slot>removeCurrentRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reload</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>reload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>entries_list</sender>
   <signal>selected(QModelIndex)</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>enableBootEntryEditor(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exit</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>about</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>showAboutBox()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>import_</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>import()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>export_()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dump_raw_efi_data</sender>
   <signal>triggered()</signal>
   <receiver>EFIBootEditor</receiver>
   <slot>dump()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>enableBootEntryEditor(QModelIndex)</slot>
  <slot>disableBootEntryEditor()</slot>
  <slot>showAboutBox()</slot>
  <slot>reload()</slot>
  <slot>save()</slot>
  <slot>export_()</slot>
  <slot>import()</slot>
  <slot>dump()</slot>
 </slots>
</ui>
